src/apps/mpx.erl

```diff
@@ -58,20 +62,20 @@ handle_client_message(_,_) -> ok.
 
 client_disconnected(Client) -> gen_server:call(?SERVER, {client_disconnected, Client}).
 
-handle_call({subscribe, #de_client{meta=Apps, data=AppData}=Client, App}, _, State) ->
+handle_call({subscribe, #de_client{meta=Apps, data=AppData}=Client, App, Data}, _, State) ->
   Reply = 
     case deliverly_server:find_handler(App) of
       false -> 
         ?D({app_not_exist, App}),
         {error, noexist};
       Handler ->
-        Res = Handler:authorize(app_client(App,Client), []),
+        Res = Handler:authorize(app_client(App,Client), Data),
         case Res of
           {error, Reason} -> 
             ?ACCESS("AUTH_FAILED ~p ~p ~p",[App, Client#de_client.path, Reason]),
             Res;
           Tuple -> 
-            #de_client{data=Data, encoder=Encoder}=NewClient = element(2, Res),
+            #de_client{data=ClientData, encoder=Encoder}=NewClient = element(2, Res),
```

Comment:
Line 78 is too long:             #de_client{data=ClientData, encoder=Encoder}=NewClient = element(2, Res),.